tools/internal_ci/helper_scripts/prepare_build_macos_rc

```diff
@@ -48,6 +48,10 @@ set -ex
 
 # cocoapods
 export LANG=en_US.UTF-8
+# pre-fetch cocoapods master repo with HEAD only
+mkdir -p ~/.cocoapods/repos
+git clone --depth 1 https://github.com/CocoaPods/Specs.git ~/.cocoapods/repos/master
+
 time pod repo update  # needed by python
```

Comment:
the comment `# needed by python` seems odd, is that a typo something that actually makes sense?

Reply:
it this is only required for ObjC tests (which would seem logical and the comment is a typo), I'd prefer this to move elsewhere (e.g. to obj-C build scripts), because there's no reason to run this for other languages.

Reply:
> is that a typo something that actually makes sense

I personally don't quite understand the reason why this needs to be here. @ericgribkoff do you know why Python needs it?

Reply:
> it this is only required for ObjC tests (which would seem logical and the comment is a typo)

Not necessarily true. Once we set up CI for C++ on iOS that needs this too. In summary, any library that runs on iOS, or on Mac and uses Cocoapods, will need this.

Reply:
Sure, but right now we don't have any such tests, so currently this is only for ObjC.

I found the PR where where the line was added 
https://github.com/grpc/grpc/pull/11636/files  - I don't know why it was added (besides trying to things to become green) and it was 2 years ago. I think it's fine to remove,